lib/improver/spotdata_new/neighbour_finding.py

```diff
@@ -0,0 +1,488 @@
+                example, the default 'mosg' will return the Met Office grid
+                attributes mosg__grid_domain, mosg__grid_type, etc.
+        """
+        self.minimum_dz = minimum_dz
+        self.land_constraint = land_constraint
+        self.search_radius = search_radius
+        self.site_coordinate_system = site_coordinate_system
+        self.site_x_coordinate = site_x_coordinate
+        self.site_y_coordinate = site_y_coordinate
+        self.site_altitude = 'altitude'
+        self.geodetic_coordinate_system = False
+        self.grid_metadata_identifier = grid_metadata_identifier
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        return ('<NeighbourSelection: land_constraint: {}, ' +
+                'minimum_dz: {}, search_radius: {}, site_coordinate_system'
+                ': {}, site_x_coordinate:{}, site_y_coordinate: {}>').format(
+                    self.land_constraint, self.minimum_dz, self.search_radius,
+                    self.site_coordinate_system.__class__,
+                    self.site_x_coordinate, self.site_y_coordinate)
+
+    def neighbour_finding_method_name(self):
+        """
+        Create a name to describe the neighbour method based on the constraints
+        provided.
+
+        Returns:
+            method_name (string):
+                A string that describes the neighbour finding method employed.
+                This is essentially a concatentation of the options.
+        """
+        method_name = '{}{}{}'.format('nearest',
+                                      '_land' if self.land_constraint else '',
+                                      '_minimum_dz' if self.minimum_dz else '')
+        return method_name
+
+    def _transform_sites_coordinate_system(self, x_points, y_points, cube):
+        """
+        Function to convert coordinate pairs that specify spot sites into the
+        coordinate system of the model from which data will be extracted.
+
+        Args:
+            x_points (np.array):
+                An array of x coordinates to be transformed in conjunction
+                with the corresponding y coordinates.
+            y_points (np.array):
+                An array of y coordinates to be transformed in conjunction
+                with the corresponding x coordinates.
+            cube (iris.cube.Cube):
+                A cube from the model from which data will be extracted. This
+                provides the coordinate system onto which the spot site's
+                coordinates should be remapped.
+        Returns:
+            np.array:
+                An array containing the x and y coordinates of the spot sites
+                in the target coordinate system, shaped as (n_sites, 2).
+        """
+        target_coordinate_system = cube.coord_system().as_cartopy_crs()
+
+        return target_coordinate_system.transform_points(
+            self.site_coordinate_system, x_points, y_points)[:, 0:2]
```

Comment:
Yes, I was wondering about the usefulness of this as a general utility. I don't believe we are using those previously moved to spatial anywhere else and thought me might well delete those when the old spotdata system goes. As you say, this is a bit bespoke because I am cutting away the returned z dimension, so I thought it should live in here. I will add something to the docstring to explain what is happening.